python/ray/serve/scripts.py

```diff
@@ -286,68 +289,191 @@ def run(
+# need to unblock SIGINT.
+import signal
+signal.pthread_sigmask(signal.SIG_UNBLOCK, {signal.SIGINT})
+
+import argparse
+import importlib
+import pathlib
+import signal
+import sys
+import time
+import yaml
+
+from ray import serve
+from ray._private.utils import import_attr
+from ray.autoscaler._private.cli_logger import cli_logger
+from ray.serve._private import api as _private_api
+from ray.serve.schema import ServeApplicationSchema
 
-    # Setting the runtime_env here will set defaults for the deployments.
-    ray.init(address=address, namespace=SERVE_NAMESPACE, runtime_env=final_runtime_env)
+parser = argparse.ArgumentParser()
+parser.add_argument('--config-or-import-path')
+parser.add_argument('--host')
+parser.add_argument('--port')
+parser.add_argument('--blocking', action='store_true')
+parser.add_argument('--gradio', action='store_true')
+args = parser.parse_args()
+
+if pathlib.Path(args.config_or_import_path).is_file():
+    config_path = args.config_or_import_path
+    cli_logger.print(f'Deploying from config file: "{config_path}".')
+
+    with open(config_path, 'r') as config_file:
+        config = ServeApplicationSchema.parse_obj(yaml.safe_load(config_file))
+    is_config = True
+else:
+    import_path = args.config_or_import_path
+    cli_logger.print(f'Deploying from import path: "{import_path}".')
+    node = import_attr(import_path)
+    is_config = False
+
+if is_config:
+    client = _private_api.serve_start(
+        detached=True,
+        http_options={
+            'host': config.host,
+            'port': config.port,
+            'location': 'EveryNode',
+        },
+    )
+else:
+    client = _private_api.serve_start(
+        detached=True,
+        http_options={'host': args.host, 'port': args.port, 'location': 'EveryNode'},
+    )
 
+try:
     if is_config:
-        client = _private_api.serve_start(
-            detached=True,
-            http_options={
-                "host": config.host,
-                "port": config.port,
-                "location": "EveryNode",
-            },
-        )
+        client.deploy_app(config, _blocking=args.gradio)
+        cli_logger.success('Submitted deploy config successfully.')
+        if args.gradio:
+            handle = serve.get_deployment('DAGDriver').get_handle()
     else:
-        client = _private_api.serve_start(
-            detached=True,
-            http_options={"host": host, "port": port, "location": "EveryNode"},
+        handle = serve.run(node, host=args.host, port=args.port)
+        cli_logger.success('Deployed Serve app successfully.')
+
+    if args.gradio:
+        from ray.serve.experimental.gradio_visualize_graph import GraphVisualizer
+
+        visualizer = GraphVisualizer()
+        visualizer.visualize_with_gradio(handle)
+    else:
+        if args.blocking:
+            while True:
+                # Block, letting Ray print logs to the terminal.
+                time.sleep(10)
+
+except KeyboardInterrupt:
```

Comment:
Instead of KeyboardInterrupt, why not just handle SIGTERM here? That way we can just use the existing SIGTERM followed by SIGKILL logic, and we can get rid of `RAY_JOB_STOP_SIGNAL`.  After all, it sounds like that's what SIGTERM is for ("politely ask a program to terminate."). I think the only reason we had KeyboardInterrupt before was because it was for a user facing CLI, but that's no longer the case when passing the script to Job Submission.

Reply:
Maybe the downside is `serve.shutdown()` might not finish within the 3 second timeout before SIGKILL?

Reply:
Hmm, I've tried just using SIGTERM instead of SIGINT, but for some reason have run into a lot of issues (serve isn't shut down correctly).